modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -550,6 +550,22 @@ impl ServerModule for Wallet {
     ) -> Result<TransactionItemAmount, WalletOutputError> {
         let output = output.ensure_v0_ref()?;
 
+        // In 0.4.0 we began preventing RBF withdrawals. Once we reach EoL support
```

Comment:
This method (`process_output`) is called both when checking submitted txes before accepting them into consensus, and when consensus being processed. Because of it I proposed we have a new method `check_output`, with a default impl that forwards the call to `process_output`. In tx submission code we'll call `check_output` instead of `process_output`. All modules except wallet will not implement new method (thus use a default fowarding to `process_output`), for wallet we will put this new "check" in it.

Reply:
Please let me know if I'm misunderstanding, but we only call `process_output` once in `process_transaction_with_dbtx`, which means we'd need to refactor so we have something like `process_transaction_with_dbtx` and `verify_transaction_with_dbtx` when we're processing the tx in consensus and when we're verifying the tx in submit tx.

To keep this PR's history easier to follow, I opened a separate draft PR to see if this approach is what you had in mind.

https://github.com/fedimint/fedimint/pull/5520

I prefer the implementation as-is with the minimal check in the implementation of `process_output` in the wallet module, but I don't have a strong opinion if you'd prefer to add the additional verify methods.

Reply:
Note after a call: It didn't occur to me that it's possible to restart syncing from scratch with the env var. It's not a best UX, but will do, and we generally don't expect anyone to hit it, so :+1: